Sources/Build/Command.link(SwiftModule).swift

```diff
@@ -26,7 +26,16 @@ extension Command {
             return [module] + module.recursiveDependencies.filter{ $0.type == .library }
         }.flatMap{ $0 as? SwiftModule }).contents
         
-        var objects = buildables.flatMap { SwiftcTool(module: $0, prefix: prefix, otherArgs: [], executable: linkerExec.asString, conf: conf).objects }
+        var objects = buildables.flatMap { module -> [AbsolutePath] in
+            let tool = SwiftcTool(module: module, prefix: prefix, otherArgs: [], executable: linkerExec.asString, conf: conf)
+            var objects = tool.objects
+            // To make executables debuggable, we need to link swiftmodule of the main executable module.
+            if module.type == .executable && conf == .debug {
+                assert(product.type == .Executable)
+                objects.append(tool.moduleOutputPath)
```

Comment:
Interesting, we should definitely have a comment around that as well, and also reference the importance of `-module-name`.

In addition, we should leave a FIXME in here for not keying this on the debug configuration --  it is valid to build with debug info in release builds, but we don't have a good way to spell that yet.